### PR TITLE
Priority sort for same day items

### PR DESCRIPTION
**Pull request by gerazov**
Moved priority sorting in the `sort_by_date_or_category` function so it gets done for same day items as well. Renamed func to `sort_by_date_or_priority_or_category`, a bit wordy but more adequate :slightly_smiling_face: 

Before:

image

After:

image

<details>
  <summary>Contents of test2.org</summary>

  ```org
* TODO [#A] Task 1
  SCHEDULED: <2022-03-27 нед>
* TODO [#B] Task 2
  SCHEDULED: <2022-03-27 нед>
* TODO [#C] Task 3
  SCHEDULED: <2022-03-27 нед>
* TODO [#A] Task 4
  SCHEDULED: <2022-03-27 нед>
* TODO [#B] Task 5
  SCHEDULED: <2022-03-27 нед>
* TODO [#C] Task 6
  SCHEDULED: <2022-03-27 нед>
```
</details>

### CODE REVIEW

**Comment by kristijanhusak** [comment]
That makes sense, thanks!

**Comment by gerazov** [comment]
:love_you_gesture:
